src/internal_events/tcp.rs

```diff
@@ -27,19 +26,34 @@ impl InternalEvent for TcpSocketConnectionEstablished {
 }
 
 #[derive(Debug)]
-pub struct TcpSocketConnectionFailed<E> {
+pub struct TcpSocketConnectionError<E> {
     pub error: E,
 }
 
-impl<E> InternalEvent for TcpSocketConnectionFailed<E>
+impl<E> InternalEvent for TcpSocketConnectionError<E>
 where
     E: std::error::Error,
 {
     fn emit_logs(&self) {
-        error!(message = "Unable to connect.", error = %self.error);
+        error!(
+            message = "Unable to connect.",
+            error = %self.error,
+            error_code = "failed_connecting",
+            error_type = error_type::WRITER_FAILED,
+            // this event is only used in sinks
```

Comment:
It's not clear to me what this comment is meant to convey. I get the proximate information but the larger context I'm still unsure of.